wallet/createtx.go

```diff
@@ -856,7 +846,7 @@ func (w *Wallet) compressWalletInternal(ctx context.Context, op errors.Op, dbtx
 			break
 		}
 		// Add the size of a wire.OutPoint
-		if msgtx.SerializeSize() > maximumTxSize {
+		if msgtx.SerializeSize() > w.chainParams.MaxTxSize {
```

Comment:
This is incorrect.  The maximum tx size is the maximum possible size of a transaction at the wire level.  The standardness, rules, however, limit it to [100000](https://github.com/decred/dcrd/blob/1a15991187b73e45fcad285b3f57faa7ec714cf9/internal/mempool/policy.go#L28) as the existing code here does.

Standardness rules do not apply to other networks which is why the existing code here only applies it to mainnet.

Reply:
It would probably make sense to add a field for the distinct standardness limit to the chain parameters struct and just set them to the correct values.  100000 for mainnet and whatever the `MaxTxSize` is for the other networks.   Adding a new field does not require a major module version bump of `chaincfg` either.

Then this code could safely use the new field.